src/TestingUtils/Microsoft.AspNetCore.Testing/src/xunit/FlakyAttribute.cs

```diff
@@ -0,0 +1,43 @@
+using System;
+using System.Collections.Generic;
+using Xunit.Sdk;
+
+namespace Microsoft.AspNetCore.Testing.xunit
+{
+    [TraitDiscoverer("Microsoft.AspNetCore.Testing.xunit.FlakyTestDiscoverer", "Microsoft.AspNetCore.Testing")]
+    [AttributeUsage(AttributeTargets.Method)]
+    public sealed class FlakyAttribute : Attribute, ITraitAttribute
+    {
+        private List<string> _queues = new List<string>() { "all" };
+
+        /// <summary>
+        /// Gets a URL to a GitHub issue tracking this flaky test.
+        /// </summary>
+        public string GitHubIssueUrl { get; }
+
+        /// <summary>
+        /// Gets or sets a list of helix queues on which this test is flaky. Defaults to <see cref="HelixQueues.All"/> indicating it is flaky on all Helix queues. See
+        /// <see cref="HelixQueues"/> for a list of valid values.
+        /// </summary>
+        public string OnHelixQueues
+        {
+            get => string.Join(";", _queues);
+            set => _queues = new List<string>(value.Split(';'));
```

Comment:
Do you need to pass in `StringSplitOptions.RemoveEmptyEntries`?

Reply:
Probably, though it's not broken without it I believe. It would just generate the trait `Flaky:Helix:=true` which we never care about :).